care/facility/api/viewsets/camera_preset.py

```diff
@@ -0,0 +1,94 @@
+from django.shortcuts import get_object_or_404
+from rest_framework.exceptions import NotFound
+from rest_framework.mixins import ListModelMixin
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.viewsets import GenericViewSet, ModelViewSet
+
+from care.facility.api.serializers.camera_preset import CameraPresetSerializer
+from care.facility.models import Asset, AssetBed, Bed, CameraPreset
+from care.users.models import User
+from care.utils.cache.cache_allowed_facilities import get_accessible_facilities
+
+
+class AssetBedCameraPresetViewSet(ModelViewSet):
+    serializer_class = CameraPresetSerializer
+    queryset = CameraPreset.objects.all().select_related(
+        "asset_bed", "created_by", "updated_by"
+    )
+    lookup_field = "external_id"
+    permission_classes = (IsAuthenticated,)
+
+    def get_asset_bed_obj(self):
+        user = self.request.user
+        queryset = AssetBed.objects.filter(
+            external_id=self.kwargs["assetbed_external_id"]
+        )
+        if user.is_superuser:
+            pass
+        elif user.user_type >= User.TYPE_VALUE_MAP["StateLabAdmin"]:
```

Comment:
Can we move this logic to use an abstracted function? like what we do in facilities.

Reply:
Did for `get_asset_bed_obj`, `get_asset_obj` and `get_bed_obj` and also updated assetbed/asset/bed viewsets to reuse the same